lib/screenplay/suppressed_predictions/suppressed_prediction.ex

```diff
@@ -0,0 +1,145 @@
+  end
+
+  def changeset(suppressed_prediction, attrs \\ %{}) do
+    places = Places.get()
+
+    suppressed_prediction
+    |> cast(attrs, [
+      :location_id,
+      :route_id,
+      :direction_id,
+      :clear_at_end_of_day
+    ])
+    |> validate_required([
+      :location_id,
+      :route_id,
+      :direction_id,
+      :clear_at_end_of_day
+    ])
+    |> validate_number(:direction_id,
+      less_than_or_equal_to: 1,
+      greater_than_or_equal_to: 0,
+      message: "Location Direction ID must be 0 or 1"
+    )
+    |> validate_edge_cases()
+    |> validate_location(places)
+    |> validate_routes(places)
+  end
+
+  defp validate_edge_cases(changeset) do
+    location_id = get_field(changeset, :location_id)
+    route_id = get_field(changeset, :route_id)
+
+    cond do
+      # JFK/UMass has the Ashmont/Braintree split, we ask for those stop ids rather than the parent id
+      location_id == "place-jfk" ->
+        add_error(
+          changeset,
+          :location_id,
+          "Please provide either the Ashmont or Braintree platform location_id: (#{@jfk_umass_ashmont_place} or #{@jfk_umass_braintree_place})"
+        )
+
+      # We don't handle separate Green Line and Silver Line branching route_ids
+      # That is populated when we pass data to Transit Data
+      # For internal use we just use "Green" or "Silver" to show all green/silver line routes
+      route_id in @green_line_routes ->
+        add_error(
+          changeset,
+          :route_id,
+          "Please provide just `Green` as the route_id for handling all Green Line routes"
+        )
+
+      true ->
+        changeset
+    end
+  end
+
+  defp validate_location(changeset, places) do
+    location_id = get_field(changeset, :location_id)
+
+    if places |> Enum.any?(&(&1.id == location_id)) do
+      changeset
+    else
+      add_error(changeset, :location_id, "PA/ESS Location `#{location_id}` does not exist")
+    end
+  end
+
+  defp validate_routes(changeset, places) do
+    location_id = get_field(changeset, :location_id)
+    route_id = get_field(changeset, :route_id)
+
+    places
+    |> Enum.find(fn place -> place.id == location_id end)
+    |> case do
+      nil ->
+        add_error(
+          changeset,
+          :route_id,
+          "PA/ESS Location `#{location_id}` does not exist"
+        )
+
+      place ->
+        if route_id in place.routes do
+          # For Silver Line, we make sure that the routes are all valid routes in Screenplay
+          # In this case, 741, 742, 743 and 746 within @valid_silver_line_routes
+          if route_id == "Silver" do
+            pa_ess_routes =
+              place.screens
+              |> Enum.filter(fn
+                %Screenplay.Places.Place.PaEssScreen{} -> true
+                _ -> false
+              end)
+              |> Enum.flat_map(& &1.routes)
+
+            if pa_ess_routes |> Enum.any?(fn route -> route in @valid_silver_line_routes end) do
```

Comment:
Just wanted to verify the logic here: do we need to worry about places that might have stops including than the 741, 742, 743, 746 but have other routes involved?

Right now I'm saying if there's a stop with _any_ route that contains one of those, allow that stop to be used. Only error out when a stop has _only_ routes outside of the of those approved ones.